### PR TITLE
[Merged by Bors] - chore: Split off and golf `ulift` field instances

### PR DESCRIPTION
**Pull request by YaelDillies**
Match https://github.com/leanprover-community/mathlib/pull/18590 and https://github.com/leanprover-community/mathlib/pull/18596

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by fpvandoorn** [comment]
Thanks!
Please update Mathlib.lean
bors d+

**Comment by eric-wieser** [comment]
Was this done without the aid of mathport?

**Comment by YaelDillies** [comment]
For the `ulift` stuff, yes. For the `nonneg` stuff, no.

File: Mathlib/Algebra/Ring/ULift.lean [hunk 1]
```diff
@@ -4,12 +4,11 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Scott Morrison
 
 ! This file was ported from Lean 3 source module algebra.ring.ulift
-! leanprover-community/mathlib commit 207cfac9fcd06138865b5d04f7091e46d9320432
+! leanprover-community/mathlib commit 207cfac9fcd06138865b5d04f7091e46d9320432 TODO
```
Comment (eric-wieser) [inline Mathlib/Algebra/Ring/ULift.lean:7]: The CI ought to reject this...

**Comment by fpvandoorn** [comment]
You still forgot a file in `Mathlib.lean`. Note that the `README` contains a script to update the file automatically (as long as you don't have untracked Lean files in your repo).

File: Mathlib/Algebra/Ring/ULift.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Ring/ULift.lean:7]: I mean, this PR was never marked as `awaiting-review`!

File: Mathlib/Algebra/Field/ULift.lean [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+instance divisionSemiring [DivisionSemiring α] : DivisionSemiring (ULift α) := by
+  refine' down_injective.divisionSemiring down _ _ _ _ _ _ _ _ _ _ <;> intros <;> rfl
+#align ulift.division_semiring ULift.divisionSemiring
+
+instance semifield [Semifield α] : Semifield (ULift α) :=
+  { ULift.divisionSemiring, ULift.commGroupWithZero with }
+#align ulift.semifield ULift.semifield
+
+instance divisionRing [DivisionRing α] : DivisionRing (ULift α) :=
+  { ULift.divisionSemiring, ULift.addGroup with }
```
Comment (YaelDillies) [inline Mathlib/Algebra/Field/ULift.lean:48]: I don't how to fix this error:
```
type mismatch
  HEq.rfl
has type
  HEq ?m.2487 ?m.2487 : Prop
but is expected to have type
  IntCast.intCast ↑n✝ = ↑n✝ : Prop
```

File: Mathlib/Algebra/Group/ULift.lean [hunk 3]
```diff
@@ -140,23 +140,47 @@ instance monoid [Monoid α] : Monoid (ULift α) :=
...
+#align ulift.comm_monoid ULift.commMonoid
+#align ulift.add_comm_monoid ULift.addCommMonoid
+
+instance natCast [NatCast α] : NatCast (ULift α) := ⟨λ a ↦ up a⟩
+#align ulift.has_nat_cast ULift.natCast
+instance intCast [IntCast α] : IntCast (ULift α) := ⟨λ a ↦ up a⟩
+#align ulift.has_int_cast ULift.intCast
+
+@[simp, norm_cast]
+theorem up_natCast [NatCast α] (n : ℕ) : up (n : α) = n :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/Group/ULift.lean:155]: Isn't it `nat_cast`?

Comment (eric-wieser) [reply Mathlib/Algebra/Group/ULift.lean:155]: Guess it depends [where your coin flip landed](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/naming.20conventions/near/342546068) so this is out of scope.

**Comment by eric-wieser** [comment]
This is forward-porting two PRs (also #18596), not one, right? Please don't do that without making it clear in the PR description.

**Comment by YaelDillies** [comment]
Sorry, my edit to the PR description seems to have been lost in the many tabs I have open.

File: Mathlib/Algebra/Order/Nonneg/Field.lean [hunk 4]
```diff
@@ -4,14 +4,14 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Floris van Doorn
 
 ! This file was ported from Lean 3 source module algebra.order.nonneg.field
-! leanprover-community/mathlib commit dc6c365e751e34d100e80fe6e314c3c3e0fd2988
+! leanprover-community/mathlib commit 13e18cfa070ea337ea960176414f5ae3a1534aae
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Nonneg/Field.lean:7]:
```suggestion
! leanprover-community/mathlib commit b3f4f007a962e3787aa0f3b5c7942a1317f7d88e
```
This sha is wrong

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**